# HG changeset patch
# User Michael Haubenwallner <haubi@gentoo.org>
# Date 1336718929 -7200
# Branch 2.7
# Node ID cbc5a08bad11edf091ca2e093eca704159efc615
# Parent  50f0a2d5801d5440cc67bf91e43911ce6f3c2d24
AIX sometimes leaves ".nfsEBD1F5" files around. Have seen this error
while installing python - listdir() might have seen it, but subsequent
copy_file() failed, because AIX removed it in the meantime.
copying build/lib.aix-5.3-2.5/dl.so -> /eprefix/var/tmp/portage/dev-lang/python-2.5.1-r5/image/eprefix/usr/lib/python2.5/lib-dynload
  error: can't copy 'build/lib.aix-5.3-2.5/.nfsEBD1F5': doesn't exist or not a regular file
  make: *** [sharedinstall] Error 1
Solaris NFS does the same.  So, just ignore files starting with ".nfs".

http://bugs.python.org/issue7719

diff -r 50f0a2d5801d -r cbc5a08bad11 Lib/distutils/dir_util.py
--- a/Lib/distutils/dir_util.py	Fri May 11 08:48:49 2012 +0200
+++ b/Lib/distutils/dir_util.py	Fri May 11 08:48:49 2012 +0200
@@ -144,7 +144,9 @@
         src_name = os.path.join(src, n)
         dst_name = os.path.join(dst, n)
 
-        if preserve_symlinks and os.path.islink(src_name):
+        if n.startswith('.nfs'):
+            pass # do not consider NFS cache (?) files
+        elif preserve_symlinks and os.path.islink(src_name):
             link_dest = os.readlink(src_name)
             if verbose >= 1:
                 log.info("linking %s -> %s", dst_name, link_dest)
